configure.ac

```diff
@@ -903,6 +903,20 @@ if test "$use_hardening" != "no"; then
       ;;
   esac
 
+  case $host in
+    *aarch64-unknown-linux*)
+      AX_CHECK_COMPILE_FLAG([-mbranch-protection=standard], [
+        SAVED_CXXFLAGS="$CXXFLAGS"
+        CXXFLAGS="$CXXFLAGS -mbranch-protection=standard"
+        AX_CHECK_LINK_FLAG([-Wl,-z,force-bti], [
+          HARDENED_CXXFLAGS="$HARDENED_CXXFLAGS -mbranch-protection=standard"
+          HARDENED_LDFLAGS="$HARDENED_LDFLAGS -Wl,-z,force-bti"
+        ], [], [$LDFLAG_WERROR])
+        AX_CHECK_LINK_FLAG([-Wl,-z,pac-plt], [HARDENED_LDFLAGS="$HARDENED_LDFLAGS -Wl,-z,pac-plt"], [], [$LDFLAG_WERROR])
+        CXXFLAGS="$SAVED_CXXFLAGS"
+      ])
+      ;;
+  esac
```

Comment:
Won't this break a normal from-source build if the C/C++ standard libraries are BTI-ready, but other dependencies are not?

Reply:
Made this Guix-only for now.